tests/abi/ABITests.cmake

```diff
@@ -2,20 +2,40 @@
 # This file is distributed under the MIT License. See LICENSE.md for details.
 #
 
+# Ensure the generator script is run only inside the `revng-qa` project
+if("${CMAKE_PROJECT_NAME}" STREQUAL "revng-qa")
+  execute_process(
+    COMMAND
+      "python3"
+      "${CMAKE_SOURCE_DIR}/scripts/abi_artifact_generator/generate.py"
+      "${CMAKE_SOURCE_DIR}/scripts/abi_artifact_generator/templates/"
+      "${CMAKE_SOURCE_DIR}/scripts/abi_artifact_generator/config/"
+      "${CMAKE_SOURCE_DIR}/tests/abi/generated/")
```

Comment:
* `execute_process` runs at configure-time. I think we want to run this at build time with `add_custom_command`/`add_custom_target`.
* Generate the source file in the build directory, not the source directory.

Reply:
1. Yeah, I'm aware, this is the side-effect of trying to deal with the complications around the second problem - so if we deal with it - we get this for free.

2. Now to the core issue. I did fix `register_artifact` to actually support multiple source files. But due to how it's structured, having those in wildly different directories still doesn't work nicely. The problem is that we'd need to provide the "common" path piece and then subpaths from it. Except if one of them is in `${CMAKE_SOURCE_DIR}` and the other one is in `${CMAKE_BUILD_DIR}` - we should not try to lump them together - there's no guaranties about how they relate to one another. I'm afraid that fixing this would require quite a substantial rework of `register_artifact` macro and also modifications to how its users provide source paths.

P. S. I'm sorry I forgot to bring this up like I intended to do when I decided to temporarily generate files in the source directory and "deal with it later".

Reply:
Done.